include/rocksdb/memory_allocator.h

```diff
@@ -34,9 +35,38 @@ class MemoryAllocator {
     // default implementation just returns the allocation size
     return allocation_size;
   }
+
+  std::string GetId() const override { return GenerateIndividualId(); }
+};
+
+class MemoryAllocatorWrapper : public MemoryAllocator {
```

Comment:
Can you explain a little why we need this "Wrapper" class?

Reply:
The CountedMemoryAllocator (a new class) extends this base one.  If you rather it not, I can eliminate this class, but then the pattern does not follow almost every other Customizable class with Base and BaseWrapper

Reply:
If it is not necessary but just keep the pattern, we may not need it. We can add in the future if it is really useful.

Reply:
Tests under table_test use the CountedMemoryAllocator.  Prior to this PR, the CountedMemoryAllocator was simply Default+count and not its own allocator.  Are you suggesting I revert that?  Is there a reason that we would not want to be able to potentially count allocations (for debug/stats purposes)?

Reply:
I prefer not to add the wrapper class to the public, it is kind of confusing. When we really need it, we can easily add in the future.